libs/core/src/main/java/org/elasticsearch/core/UpdateBeforeV9.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.core;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to identify a block of code (a whole class, a method, or a field) that needs to be reviewed (for cleanup, remove or change)
+ * before releasing 9.0
+ */
+@Retention(RetentionPolicy.SOURCE)
+@Target({ ElementType.CONSTRUCTOR, ElementType.FIELD, ElementType.METHOD, ElementType.TYPE })
+public @interface UpdateBeforeV9 {
```

Comment:
Not sure about the name. `UpdateBeforeV9` implies this just needs to be modified at some point before v9 is released. But these need to be modified _on the v9 branch when it happens_. Maybe `UpdateForV9` or `DeprecatedForV9` instead?